src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -67,113 +69,143 @@ public static ParseUDFsResult ParseUDFsScript(string script, CultureInfo loc = n
+        /// <param name="tid"></param>
+        /// <returns></returns>
+        private Token TokPlainEat(TokKind tid)
+        {
+            if (_curs.TidCur == tid)
+            {
+                return _curs.TokMove();
+            }
+
+            return null;
+        }
+
         private ParseUDFsResult ParseUDFs(string script)
         {
             var udfs = new List<UDF>();
 
+            // <root> ::= (<udf> ';')*
             while (_curs.TokCur.Kind != TokKind.Eof)
             {
-                //var args = new HashSet<UDFArg>();
-                var args = new Dictionary<string, UDFArg>();
                 ParseTrivia();
-
-                // Store the name of the UDF.
-                var ident = TokEat(TokKind.Ident);
-                if (ident == null)
+                if (!ParseUDF(udfs))
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingIdentifier);
-                    break;
+                    return new ParseUDFsResult(udfs, _errors);
                 }
 
-                if (TokEat(TokKind.ParenOpen) == null)
+                ParseTrivia();
+                
+                if (TokEat(TokKind.Semicolon) == null)
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingOpenParen);
                     break;
                 }
+            }
 
-                ParseTrivia();
-
-                // Parse arguments.
-                while (_curs.TokCur.Kind != TokKind.ParenClose)
-                {
-                    ParseTrivia();
-                    var varIdent = TokEat(TokKind.Ident);
-                    ParseTrivia();
-                    if (TokEat(TokKind.Colon) == null)
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingColon);
-                        break;
-                    }
-
-                    ParseTrivia();
-                    var varType = TokEat(TokKind.Ident);
-                    ParseTrivia();
-                    if (args.ContainsKey(varIdent.ToString()))
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_DuplicateArgName);
-                        return new ParseUDFsResult(udfs, _errors);
-                    }
+            return new ParseUDFsResult(udfs, _errors);
+        }
 
-                    args.Add(varIdent.ToString(), new UDFArg(varIdent.As<IdentToken>(), varType.As<IdentToken>()));
-                    if (_curs.TokCur.Kind != TokKind.ParenClose && _curs.TokCur.Kind != TokKind.Comma)
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingComma);
-                        return new ParseUDFsResult(udfs, _errors);
-                    }
-                    else if (_curs.TokCur.Kind == TokKind.Comma)
-                    {
-                        TokEat(TokKind.Comma);
-                    }
-                }
+        private bool ParseUDF(List<UDF> udfs)
```

Comment:
probably should be `bool TryParseUdf(out UDF)`, and the caller checks for collisions instead of the parser on line 140. 
basically, it's still valid to parse something that collides